### DISCUSSION TITLE
Who is in charge of freeing the result of VImage::get_blob and if it's the caller, how?

### DISCUSSION THREAD

**Discussion by jaubourg** [libvips/libvips#2012] [Q&A]
The documentation is pretty sparse on the subject but `VImage::get_blob` returns a `const void *` or so seems to imply gcc. As such, it would mean the memory shouldn't be freed but it seems a bit weird to me.

Who has ownership of the returned value? If it's the caller, how do we free it? Using `g_free`? Do I need to copy the memory block myself and not to rely on the original `const void *` in case it gets freed at one point?

**Comment by jcupitt** [comment]
Hello @jaubourg,

The blob keeps ownership, so the pointer from `get_blob()` is valid for as long as you hold a reference to the blob. The docs have a note about this:

https://libvips.github.io/libvips/API/current/libvips-type.html#vips-blob-get

You can see the `transfer-none` tag on the result.

Yes, you need to take a copy if you want to keep the value around. Blobs are reference-counted, so you can't really steal the libvips value (it might be in use elsewhere).

**Comment by jaubourg** [comment]
Hey @jcupitt ,

It seems to make sense but since I never interact with the VipsBlob directly (everything is encapsulated in VImage from my code's POV) I have to admit determining what's movable and what's not is _pretty_ tricky.

Thanks for the quick answer, it's highly appreciated!
